homeassistant/components/lovelace/__init__.py

```diff
@@ -57,40 +82,116 @@ def save_yaml(fname: str, data: JSON_TYPE):
     except YAMLError as exc:
-        _LOGGER.error("YAML error: %s", exc)
+        _LOGGER.error("YAML error in %s: %s", fname, exc)
         raise HomeAssistantError(exc)
     except UnicodeDecodeError as exc:
         _LOGGER.error("Unable to read file %s: %s", fname, exc)
         raise HomeAssistantError(exc)
 
 
 def load_config(fname: str) -> JSON_TYPE:
-    """Load a YAML file and adds id to card if not present."""
+    """Load a YAML file and adds id to views and cards if not present."""
     config = load_yaml(fname)
-    # Check if all cards have an ID or else add one
+    # Check if all views and cards have an id or else add one
     updated = False
+    index = 0
     for view in config.get('views', []):
+        if 'id' not in view:
+            updated = True
+            view.insert(0, 'id', index,
+                        comment="Automatically created id")
         for card in view.get('cards', []):
             if 'id' not in card:
                 updated = True
-                card['id'] = uuid.uuid4().hex
-                card.move_to_end('id', last=False)
+                card.insert(0, 'id', uuid.uuid4().hex,
+                            comment="Automatically created id")
+        index += 1
     if updated:
         save_yaml(fname, config)
     return config
 
 
+def object_to_yaml(data: JSON_TYPE) -> str:
+    """Create yaml string from object."""
+    from ruamel.yaml import YAML
+    from ruamel.yaml.error import YAMLError
+    from ruamel.yaml.compat import StringIO
+    yaml = YAML(typ='rt')
+    yaml.indent(sequence=4, offset=2)
+    stream = StringIO()
+    try:
+        yaml.dump(data, stream)
+        return stream.getvalue()
+    except YAMLError as exc:
+        _LOGGER.error("YAML error: %s", exc)
+        raise HomeAssistantError(exc)
+
+
+def yaml_to_object(data: str) -> JSON_TYPE:
+    """Create object from yaml string."""
+    from ruamel.yaml import YAML
+    from ruamel.yaml.error import YAMLError
+    yaml = YAML(typ='rt')
+    try:
+        return yaml.load(data)
+    except YAMLError as exc:
+        _LOGGER.error("YAML error: %s", exc)
+        raise HomeAssistantError(exc)
+
+
+def get_card(fname: str, card_id: str, data_format: str) -> JSON_TYPE:
+    """Load a specific card config for id."""
+    config = load_yaml(fname)
+    for view in config.get('views', []):
+        for card in view.get('cards', []):
+            if card.get('id') == card_id:
+                if data_format == 'yaml':
+                    return object_to_yaml(card)
+                return card
+
+    raise CardNotFoundError(
+        "Card with ID: {} was not found in {}.".format(card_id, fname))
+
+
+def set_card(fname: str, card_id: str, card_config: str, data_format: str)\
+        -> bool:
+    """Save a specific card config for id."""
+    config = load_yaml(fname)
+    for view in config.get('views', []):
+        for card in view.get('cards', []):
+            if card.get('id') == card_id:
```

Comment:
you can reverse this statement to save an indentation and make code easier to read:

```python
if card.get('id') != card_id:
    continue
```